Sources/SwiftParser/TopLevel.swift

```diff
@@ -189,6 +189,35 @@ extension Parser {
     )
   }
 
+  private mutating func parseStatementItem() -> RawCodeBlockItemSyntax.Item {
+    let stmt = self.parseStatement()
+
+    // Special case: An 'if' or 'switch' statement followed by an 'as' must
+    // be an if/switch expression in a coercion.
+    // We could also achieve this by more eagerly attempting to parse an 'if'
+    // or 'switch' as an expression when in statement position, but that
+    // could result in less useful recovery behavior.
+    if at(.keyword(.as)), let stmtExpr = stmt.as(RawExpressionStmtSyntax.self) {
+      let expr = stmtExpr.expression
```

Comment:
Can be shortened slightly

```suggestion
    if at(.keyword(.as)), let expr = stmt.as(RawExpressionStmtSyntax.self).expression {
```